index-migration/indexMigration.py

```diff
@@ -0,0 +1,66 @@
+import subprocess
+import time
+
+def check_directory(filename):
+    """
+    To verify if filename is in current directory
+    """
+    ls_out=subprocess.Popen(['ls'], stdout=subprocess.PIPE)
+    directories = ls_out.stdout.read()
+    return filename in directories
+
+def backup_db():
+    """
+    To backup mongo databases and save it to a directory called dump.bak
+    """
+    if check_directory("dump"):
+       subprocess.Popen(['rm -rf dump*'], shell=True)
+       time.sleep(1)
+       assert not check_directory("dump"), "The dump file is not deleted successfully"
+
+    subprocess.Popen(['mongodump'], stdout=subprocess.PIPE)
+    time.sleep(3)
+    subprocess.Popen(['cp', '-rf', 'dump', 'dump.bak'], stdout=subprocess.PIPE)
+    ls_out=subprocess.Popen(['ls'], stdout=subprocess.PIPE)
```

Comment:
I think this will work the way you've written it, but it would probably be better to use subprocess.check_output which will raise if the process returns a non-zero exit code and will return stdout.